dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/harvest/OAIRecordRepository.java

```diff
@@ -23,6 +30,47 @@ public OAIRecordRepository() {
             .getResultList();
     }
     
+    public List<OAIRecord> findBySetNameAndRemoved(String setName, boolean removed) {
+        return em.createQuery("SELECT r from OAIRecord r where r.setName = :setName and r.removed = :removed", OAIRecord.class)
+            .setParameter("setName", setName)
+            .setParameter("removed", removed)
+            .getResultList();
+    }
+    
+    public List<OAIRecord> findByGlobalId(String globalId) {
+        return em.createQuery("SELECT r from OAIRecord r where r.globalId = :globalId", OAIRecord.class)
+                .setParameter("globalId", globalId)
+                .getResultList();
+    }
+    
+    public List<OAIRecord> findByGlobalIds(List<String> globalIds) {
+        return em.createQuery("SELECT r from OAIRecord r where r.globalId in :globalId", OAIRecord.class)
```

Comment:
Yes. Very good catch. It was last minute refactor and I didn,'t notice that. I will also add integration test for this method so it will be easier to spot if something is wrong